lib_eio/clock/clock_stubs.c

```diff
@@ -0,0 +1,151 @@
+#include <caml/alloc.h>
+#include <caml/mlvalues.h>
+#include <caml/fail.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "config.h"
+
+#define OCAML_EIO_RAISE_SYS_ERROR(ERR)                               \
+  do { caml_raise_sys_error (caml_copy_string("eio.clock: " ERR)); } \
+  while (0)
```

Comment:
This looks similar to https://github.com/dbuenzli/mtime/blob/master/src-clock/mtime_clock_stubs.c, but there are also lots of differences. Is it based on that? What was the reason for the changes? Should we just depend on mtime?

Reply:
Not sure. It certainly breaks backward compatibility on linux and macos. Linux tried to correct the CLOCK_MONOTONIC behavior to the - now - correct CLOCK_BOOTTIME. But it got reverted due to backwards compatibility issues. 

@dbuenzli Are you open to changing the behaviour of mtime to match the clock source as we have done here?

Reply:
Adapting this PR to use mtime will have to be in another PR in the future after mtime is updated.